Compiling with kf5-syntax-highlighting-devel-5.91.0

## Environment Details
Hello,
kf5-syntax-highlighting-devel between version 5.90.0 and 5.91.0 dropped in the cmake files advertising of the /usr/include/KF5 include 
directory. 

https://bugzilla.redhat.com/show_bug.cgi?id=2059619

iaito version 5.5.0 commit 28a1099603b3fa671bfbb226025d1a8c45558471

I have used this patch to make iaito compile with the new version of syntax highlighting

diff -ru iaito-28a1099603b3fa671bfbb226025d1a8c45558471/src/CMakeLists.txt iaito-28a1099603b3fa671bfbb226025d1a8c45558471.new/src/CMakeLists.txt
--- iaito-28a1099603b3fa671bfbb226025d1a8c45558471/src/CMakeLists.txt	2022-03-01 17:11:21.973408404 +0100
+++ iaito-28a1099603b3fa671bfbb226025d1a8c45558471.new/src/CMakeLists.txt	2022-03-03 00:58:01.088823994 +0100
@@ -269,6 +269,7 @@
 if(TARGET KF5::SyntaxHighlighting)
     target_link_libraries(iaito PRIVATE KF5::SyntaxHighlighting)
     target_compile_definitions(iaito PRIVATE IAITO_ENABLE_KSYNTAXHIGHLIGHTING)
+    include_directories(AFTER /usr/include/KF5)
 endif()
 
 if (IAITO_APPIMAGE_BUILD)


====== Build log
Actual results:
/usr/bin/g++ -DIAITO_ENABLE_GRAPHVIZ -DIAITO_ENABLE_KSYNTAXHIGHLIGHTING -DIAITO_SOURCE_BUILD -DQT_CORE_LIB -DQT_GUI_LIB -DQT_NETWORK_LIB -DQT_NO_DEBUG -DQT_SVG_LIB -DQT_WIDGETS_LIB -Diaito_EXPORTS -I/builddir/build/BUILD/iaito-28a1099603b3fa671bfbb226025d1a8c45558471/redhat-linux-build -I/builddir/build/BUILD/iaito-28a1099603b3fa671bfbb226025d1a8c45558471/src -I/builddir/build/BUILD/iaito-28a1099603b3fa671bfbb226025d1a8c45558471/redhat-linux-build/iaito_autogen/include -I/builddir/build/BUILD/iaito-28a1099603b3fa671bfbb226025d1a8c45558471/src/core -I/builddir/build/BUILD/iaito-28a1099603b3fa671bfbb226025d1a8c45558471/src/widgets -I/builddir/build/BUILD/iaito-28a1099603b3fa671bfbb226025d1a8c45558471/src/common -I/builddir/build/BUILD/iaito-28a1099603b3fa671bfbb226025d1a8c45558471/src/plugins -I/builddir/build/BUILD/iaito-28a1099603b3fa671bfbb226025d1a8c45558471/src/menus -I/builddir/build/BUILD/iaito-28a1099603b3fa671bfbb226025d1a8c45558471/src/. -isystem /usr/include/graphviz -isystem /usr/include/qt5 -isystem /usr/include/qt5/QtCore -isystem /usr/lib64/qt5/mkspecs/linux-g++ -isystem /usr/include/qt5/QtWidgets -isystem /usr/include/qt5/QtGui -isystem /usr/include/qt5/QtSvg -isystem /usr/include/qt5/QtNetwork -isystem /usr/include/libr -isystem /usr/include/libr/sdb -isystem /usr/include/capstone -isystem /usr/include/KF5/KSyntaxHighlighting -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -m64  -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -fvisibility=hidden   -Wall -Wextra -fPIC -std=gnu++11 -MD -MT CMakeFiles/iaito.dir/common/SyntaxHighlighter.cpp.o -MF CMakeFiles/iaito.dir/common/SyntaxHighlighter.cpp.o.d -o CMakeFiles/iaito.dir/common/SyntaxHighlighter.cpp.o -c /builddir/build/BUILD/iaito-28a1099603b3fa671bfbb226025d1a8c45558471/src/common/SyntaxHighlighter.cpp
/builddir/build/BUILD/iaito-28a1099603b3fa671bfbb226025d1a8c45558471/src/common/Configuration.cpp:10:10: fatal error: KSyntaxHighlighting/repository.h: No such file or directory
   10 | #include <KSyntaxHighlighting/repository.h>
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
gmake[2]: *** [CMakeFiles/iaito.dir/build.make:901: CMakeFiles/iaito.dir/common/Configuration.cpp.o] Error 1
gmake[2]: *** Waiting for unfinished jobs....
In file included from /builddir/build/BUILD/iaito-28a1099603b3fa671bfbb226025d1a8c45558471/redhat-linux-build/iaito_autogen/P66WHCG54J/moc_SyntaxHighlighter.cpp:10,
                 from /builddir/build/BUILD/iaito-28a1099603b3fa671bfbb226025d1a8c45558471/redhat-linux-build/iaito_autogen/mocs_compilation.cpp:24:
/builddir/build/BUILD/iaito-28a1099603b3fa671bfbb226025d1a8c45558471/redhat-linux-build/iaito_autogen/P66WHCG54J/../../../src/common/SyntaxHighlighter.h:13:10: fatal error: KSyntaxHighlighting/syntaxhighlighter.h: No such file or directory
   13 | #include <KSyntaxHighlighting/syntaxhighlighter.h>
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
In file included from /builddir/build/BUILD/iaito-28a1099603b3fa671bfbb226025d1a8c45558471/src/common/SyntaxHighlighter.cpp:2:
/builddir/build/BUILD/iaito-28a1099603b3fa671bfbb226025d1a8c45558471/src/common/SyntaxHighlighter.h:13:10: fatal error: KSyntaxHighlighting/syntaxhighlighter.h: No such file or directory
   13 | #include <KSyntaxHighlighting/syntaxhighlighter.h>
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
gmake[2]: *** [CMakeFiles/iaito.dir/build.make:313: CMakeFiles/iaito.dir/iaito_autogen/mocs_compilation.cpp.o] Error 1
gmake[2]: *** [CMakeFiles/iaito.dir/build.make:971: CMakeFiles/iaito.dir/common/SyntaxHighlighter.cpp.o] Error 1
In file included from /builddir/build/BUILD/iaito-28a1099603b3fa671bfbb226025d1a8c45558471/src/widgets/DisassemblerGraphView.cpp:9:
/builddir/build/BUILD/iaito-28a1099603b3fa671bfbb226025d1a8c45558471/src/common/SyntaxHighlighter.h:13:10: fatal error: KSyntaxHighlighting/syntaxhighlighter.h: No such file or directory
   13 | #include <KSyntaxHighlighting/syntaxhighlighter.h>
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
gmake[2]: *** [CMakeFiles/iaito.dir/build.make:383: CMakeFiles/iaito.dir/widgets/DisassemblerGraphView.cpp.o] Error 1
In file included from /builddir/build/BUILD/iaito-28a1099603b3fa671bfbb226025d1a8c45558471/src/widgets/HexdumpWidget.cpp:7:
/builddir/build/BUILD/iaito-28a1099603b3fa671bfbb226025d1a8c45558471/src/common/SyntaxHighlighter.h:13:10: fatal error: KSyntaxHighlighting/syntaxhighlighter.h: No such file or directory
   13 | #include <KSyntaxHighlighting/syntaxhighlighter.h>
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
gmake[2]: *** [CMakeFiles/iaito.dir/build.make:887: CMakeFiles/iaito.dir/widgets/HexdumpWidget.cpp.o] Error 1
/builddir/build/BUILD/iaito-28a1099603b3fa671bfbb226025d1a8c45558471/src/widgets/RelocsWidget.cpp: In member function 'virtual int RelocsModel::rowCount(const QModelIndex&) const':
/builddir/build/BUILD/iaito-28a1099603b3fa671bfbb226025d1a8c45558471/src/widgets/RelocsWidget.cpp:14:46: warning: unused parameter 'parent' [-Wunused-parameter]
   14 | int RelocsModel::rowCount(const QModelIndex &parent) const
      |                           ~~~~~~~~~~~~~~~~~~~^~~~~~
/builddir/build/BUILD/iaito-28a1099603b3fa671bfbb226025d1a8c45558471/src/widgets/ImportsWidget.cpp: In member function 'virtual int ImportsModel::rowCount(const QModelIndex&) const':
/builddir/build/BUILD/iaito-28a1099603b3fa671bfbb226025d1a8c45558471/src/widgets/ImportsWidget.cpp:17:47: warning: unused parameter 'parent' [-Wunused-parameter]
   17 | int ImportsModel::rowCount(const QModelIndex &parent) const
      |                            ~~~~~~~~~~~~~~~~~~~^~~~~~
/builddir/build/BUILD/iaito-28a1099603b3fa671bfbb226025d1a8c45558471/src/widgets/ExportsWidget.cpp: In member function 'virtual bool ExportsProxyModel::lessThan(const QModelIndex&, const QModelIndex&) const':
/builddir/build/BUILD/iaito-28a1099603b3fa671bfbb226025d1a8c45558471/src/widgets/ExportsWidget.cpp:122:9: warning: this statement may fall through [-Wimplicit-fallthrough=]
  122 |         if (leftExp.type != rightExp.type)
      |         ^~
/builddir/build/BUILD/iaito-28a1099603b3fa671bfbb226025d1a8c45558471/src/widgets/ExportsWidget.cpp:124:5: note: here
  124 |     case ExportsModel::CommentColumn:
      |     ^~~~
/builddir/build/BUILD/iaito-28a1099603b3fa671bfbb226025d1a8c45558471/src/widgets/ImportsWidget.cpp: In member function 'virtual bool ImportsProxyModel::lessThan(const QModelIndex&, const QModelIndex&) const':
/builddir/build/BUILD/iaito-28a1099603b3fa671bfbb226025d1a8c45558471/src/widgets/ImportsWidget.cpp:144:9: warning: this statement may fall through [-Wimplicit-fallthrough=]
  144 |         if (leftImport.libname != rightImport.libname)
      |         ^~
/builddir/build/BUILD/iaito-28a1099603b3fa671bfbb226025d1a8c45558471/src/widgets/ImportsWidget.cpp:147:5: note: here
  147 |     case ImportsModel::NameColumn:
      |     ^~~~
gmake[2]: Leaving directory '/builddir/build/BUILD/iaito-28a1099603b3fa671bfbb226025d1a8c45558471/redhat-linux-build'
gmake[1]: *** [CMakeFiles/Makefile2:89: CMakeFiles/iaito.dir/all] Error 2
gmake[1]: Leaving directory '/builddir/build/BUILD/iaito-28a1099603b3fa671bfbb226025d1a8c45558471/redhat-linux-build'
gmake: *** [Makefile:159: all] Error 2

--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -268,6 +268,8 @@ endif()
 if(TARGET KF5::SyntaxHighlighting)
     target_link_libraries(iaito PRIVATE KF5::SyntaxHighlighting)
     target_compile_definitions(iaito PRIVATE IAITO_ENABLE_KSYNTAXHIGHLIGHTING)
+    # in the kf5-syntax-highlighting-devel 5.91.0 the cmake files dropped advertising of the /usr/include/KF5 include directory
+    include_directories(AFTER /usr/include/KF5)
 endif()
 
 if (IAITO_APPIMAGE_BUILD)

